protocol-models/src/main/java/io/airbyte/protocol/models/CatalogHelpers.java

```diff
@@ -378,8 +378,13 @@ public static Set<StreamTransform> getCatalogDiff(final AirbyteCatalog oldCatalo
               .findFirst();
 
           if (!streamOld.equals(streamNew)) {
-            streamTransforms.add(StreamTransform.createUpdateStreamTransform(descriptor,
-                getStreamDiff(streamOld, streamNew, stream)));
+            // getStreamDiff only checks for differences in the stream's field name or field type
+            // but there are a number of reasons the streams might be different (such as a source-defined
+            // primary key or cursor changing).  These should not be expressed as "stream updates".
+            UpdateStreamTransform streamTransform =getStreamDiff(streamOld, streamNew, stream);
```

Comment:
linting will address the spacing.

Reply:
Thanks!  Ran formatting + added a test for good measure.